### PR TITLE
Example project

### PR DESCRIPTION
**Pull request by Kellen-Sullivan**
Add load data file to week 2

### CODE REVIEW

File: example_project/week_1/load_student_data.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+def main():
+    """Load the student dataset, print summary information, and plot graduation status distribution."""
+
+    # Fetch students dataset from ucirepo
+    dataset = fetch_ucirepo(id=697) 
+    
+    # Store the dataset in two separate pandas dataframes
+    # X = all columns (features) describing students (i.e., age, financial status)
+    # y = graduation status of each student: Graduate, Dropout, or Enrolled
```
Comment (npragin) [inline example_project/week_1/load_student_data.py:27]: Great!

File: example_project/week_2/exploratory_data_analysis.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import pandas as pd
+
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import KFold, cross_val_score, train_test_split
+from sklearn.tree import DecisionTreeClassifier
+from ucimlrepo import fetch_ucirepo
+from ydata_profiling import ProfileReport
+
+
+def feature_engineering(X: pd.DataFrame) -> pd.DataFrame:
```
Comment (npragin) [inline example_project/week_2/exploratory_data_analysis.py:19]: Great comments on this function discussing your thought process on feature engineering

File: example_project/week_2/exploratory_data_analysis.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    cv_scores = cross_val_score(model, X_train, y_train, cv=kf, scoring="accuracy")
+
+    print("Cross-Validation Scores:", cv_scores)
+    print(f"Mean Validation Score: {np.mean(cv_scores)}")
+
+    # Update dataset with new features
+    X = feature_engineering(X)
+
+    # Split updated dataset into train and test subsets (80% train - 20% test)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
```
Comment (npragin) [inline example_project/week_2/exploratory_data_analysis.py:79]: Why are we splitting for a second time?

File: example_project/week_2/exploratory_data_analysis.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    print("Cross Validation scores:", cv_scores)
+    print(f"Mean Validation Score: {np.mean(cv_scores)}\n")
+
+    # Train model on entire training set
+    model.fit(X_train, y_train)
+
+    # Evaluate model on test set
+    predictions = model.predict(X_test)
+    accuracy = accuracy_score(predictions, y_test)
+    print(f"Accuracy Score on Test Data: {accuracy}")
```
Comment (npragin) [inline example_project/week_2/exploratory_data_analysis.py:94]: We shouldn't be evaluating on test data in this file. Whenever we use performance on data to make decisions about our model/data, we should use validation data or CV to get those metrics for comparison.

File: example_project/week_2/exploratory_data_analysis.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    # Split updated dataset into train and test subsets (80% train - 20% test)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+    # Use 4-Fold Cross-Validation to re-evaluate model performance after feature engineering
+    cv_scores = cross_val_score(model, X_train, y_train, cv=kf, scoring="accuracy")
+
+    print("\nScores with Feature Engineering: ")
+    print("Cross Validation scores:", cv_scores)
+    print(f"Mean Validation Score: {np.mean(cv_scores)}\n")
+
```
Comment (npragin) [inline example_project/week_2/exploratory_data_analysis.py:87]: Consider leaving a comment at the end along the lines of, "Ideally, we would evaluate performance with and without each engineered feature and use the ones that improve performance."

File: example_project/week_2/exploratory_data_analysis.py (hunk 3 above)
Comment (Kellen-Sullivan) [reply example_project/week_2/exploratory_data_analysis.py:79]: I split for a second time because I believe if I don't then the X_train data won't have the updated features.

**Comment by Kellen-Sullivan** [comment]
So if we are not using the test data in this file, then I probably don't need to use train_test_split right? I could just call
cv_scores = cross_val_score(model, X, y, cv=kf, scoring="accuracy")

Or is it important to still have that hold out data even though we aren't using it?

File: example_project/week_2/exploratory_data_analysis.py (hunk 5 above)
Comment (Kellen-Sullivan) [reply example_project/week_2/exploratory_data_analysis.py:87]: will do!

File: example_project/week_2/exploratory_data_analysis.py (hunk 3 above)
Comment (npragin) [reply example_project/week_2/exploratory_data_analysis.py:79]: My bad! You are absolutely right

**Comment by npragin** [comment]
> So if we are not using the test data in this file, then I probably don't need to use train_test_split right? I could just call cv_scores = cross_val_score(model, X, y, cv=kf, scoring="accuracy")
> 
> This would address the having to split train and test data twice issue, but it might still be important to have the hold out data even though we aren't using it to show the standard practice. What are your thoughts on this.

We still need to call `train_test_split`, otherwise we will be using our test data during the validation process. You pointed this out correctly in the second piece of your message.
